src/main/java/seedu/address/logic/commands/DeleteTagCommand.java

```diff
@@ -31,77 +28,107 @@ public class DeleteTagCommand extends UndoableCommand {
 
-    public static final String MESSAGE_SUCCESS = "Tag(s) deleted.";
-    public static final String MESSAGE_NONEXISTENT = "Some tag(s) provided have not been added before.\n";
-    public static final String MESSAGE_CONFIRMATION = "The following tags are tagged on some guests:\n"
-            + "%s\n"
-            + "Deleting the tags will remove them from the guests.\n"
-            + "Are you sure you want to delete? Click 'OK' to confirm.";
-
-    public static final String MESSAGE_CANCELLED = "Deletion has been cancelled.";
-    private final Map<Tag, Set<Person>> deletedSet = new HashMap<>();
+    public static final String MESSAGE_SUCCESS = "Deleted successfully.";
+    public static final String MESSAGE_SUCCESS_FORCE = "Force deleted successfully.";
+    public static final String MESSAGE_TAGS_IN_USE = "The following tags are in use and cannot be deleted:\n%s\n"
+            + "Use 'deletetag -force' to delete tags that are in use.";
+    public static final String MESSAGE_NONEXISTENT = "tag(s) have not been added before.\n";
 
     private final List<Tag> tags;
+    private final Map<Tag, Set<Person>> deletedSet = new HashMap<>();
+    private final boolean isForceDelete;
 
     /**
      * Constructs a DeleteTagCommand to delete the specified tags.
      *
      * @param tags The tags to be deleted.
      */
-    public DeleteTagCommand(List<Tag> tags) {
+    public DeleteTagCommand(List<Tag> tags, boolean isForceDelete) {
         requireAllNonNull(tags);
         this.tags = tags;
+        this.isForceDelete = isForceDelete;
     }
 
-    /**
-     * Removes the specified tags from all persons who are tagged with these tags.
-     *
-     * @param model Model that contains the list of persons to remove tags from.
-     */
-    private void removeTagsFromPersons(Model model) {
-        for (Tag tag : tags) {
-            deletedSet.put(tag, model.removeTagFromPersons(tag));
-        }
-    }
-
-    /**
-     * Creates and returns a CommandResult based on the success of the deletion.
-     *
-     * @param isSuccessful Boolean indicating if the deletion was successful.
-     * @return CommandResult indicating the outcome of the deletion attempt.
-     */
-    private CommandResult createCommandResult(boolean isSuccessful) {
-        if (!isSuccessful) {
-            return new CommandResult(MESSAGE_NONEXISTENT);
-        }
-        return new CommandResult(MESSAGE_SUCCESS);
-    }
-
-    /**
-     * Executes the DeleteTagCommand by deleting the specified tags from the tag list in the model.
-     * If the tags are currently being used to tag any persons, the user is prompted for confirmation.
-     *
-     * @param model Model containing the tags and persons to operate on.
-     * @return CommandResult indicating the result of the command.
-     * @throws CommandException If the user cancels the deletion confirmation.
-     */
     @Override
     public CommandResult execute(Model model) throws CommandException {
```

Comment:
This method seems quite long (methods are recommended to be ~30 lines long). Is it possible to use separate methods to handle each part of the logic? ie. showErrorMessage() and handleForceDelete() can be independent methods